src/contracts/PreAMMBatcher.sol

```diff
@@ -418,4 +418,37 @@ contract PreAMMBatcher {
         }
         return newOrders;
     }
+
+    enum Direction {Ascending, Descending}
+
+    function isSortedByLimitPrice(Order[] memory orders, Direction direction)
+        internal
+        pure
+        returns (bool)
+    {
+        if (orders.length < 2) {
+            // All order sets with less than 2 elements are tautologically sorted.
+            return true;
+        }
+        for (uint256 i = 0; i < orders.length - 1; i++) {
+            Order memory orderA = orders[i];
```

Comment:
nit: i'd be curious to see how much we'd save if we kept the previous order in a variable scoped outside of the loop instead of double loading i/i+1 in each iteration